src/shared/install.c

```diff
@@ -1840,7 +1839,7 @@ int unit_file_mask(
                 if (!path)
                         return -ENOMEM;
 
-                q = create_symlink(&paths, "/dev/null", path, force, changes, n_changes);
+                q = create_symlink(&paths, "/dev/null", path, flags & UNIT_FILE_FORCE, changes, n_changes);
```

Comment:
the other option might be to also change create_symlink() to not use bools for flags anymore but just make it take the flags param directly.